publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -265,31 +268,59 @@ private Resource updateMetadata(UpdatePublicationRequest input,
         validateRequest(identifierInPath, input);
 
         var existingPublication = existingResource.toPublication();
-
-        permissionStrategy.authorize(UPDATE);
-        authorizeFileEntries(existingResource, userInstance, getModifiedFiles(existingResource, input));
-
         var publicationUpdate = input.generatePublicationUpdate(existingPublication);
 
         var customerApiClient = getCustomerApiClient();
         var customer = fetchCustomerOrFailWithBadGateway(customerApiClient, publicationUpdate.getPublisher().getId());
 
+        permissionStrategy.authorize(UPDATE);
+        authorizeFileEntries(existingResource, userInstance, getExistingFilesToUpdate(existingResource, input));
+
+        var files = Resource.fromPublication(publicationUpdate).getFiles();
+        for (File file : files) {
+            if (canNotUpdateFileToPendingOpenFile(file, customer, existingResource, userInstance)) {
+                throw new ForbiddenException();
+            }
+        }
         setRrsOnFiles(publicationUpdate, existingPublication, customer, userInstance.getUsername(), permissionStrategy);
+
         new PublishingRequestResolver(resourceService, ticketService, userInstance, customer)
             .resolve(existingPublication, publicationUpdate);
 
         return Resource.fromPublication(publicationUpdate).update(resourceService, userInstance);
     }
 
-    private static void authorizeFileEntries(Resource resource, UserInstance userInstance, List<FileEntry> modifiedFiles)
+    private static boolean canNotUpdateFileToPendingOpenFile(File file, Customer customer, Resource resource,
```

Comment:
I think I like something like this better, even if its results in a ! in the code earlier on

```java
private static boolean canUpdateFileToOpenFile(File file, Customer customer, Resource resource,
                                               UserInstance userInstance) {
    var existingFile = resource.getFileByIdentifier(file.getIdentifier());
    return existingFile.isEmpty()
           || existingFile.get().getArtifactType().equals(file.getArtifactType())
           || !(file instanceof PendingOpenFile)
           || customerAllowsOpenFiles(customer, resource)
           || canUpdateResource(userInstance);
}
```